internal/apps/devflow/flow/flow.service.go

```diff
@@ -582,37 +554,51 @@ func (s *Service) GetDevFlowInfo(ctx context.Context, req *pb.GetDevFlowInfoRequ
 			appTempBranchCommitMap[appID] = make(map[string]*apistructs.Commit)
 			locker.Unlock()
 			for _, v := range appMergeInfoMap[appID] {
-				tempBranch := targetBranchTempBranchMap[v.TargetBranch]
-				if !v.IsJoinTempBranch {
-					continue
-				}
-				baseCommit, err := s.p.bdl.GetMergeBase(apis.GetUserID(ctx), apistructs.GittarMergeBaseRequest{
-					SourceBranch: v.SourceBranch,
-					TargetBranch: tempBranch,
-					AppID:        appDto.ID,
-				})
+				repoPath := gittarPrefixOpenApi + appDto.ProjectName + "/" + appDto.Name
+				tempBranch, err := s.getTempBranchFromFlowRule(ctx, appDto.ProjectID, &v)
 				if err != nil {
 					return err
 				}
+				var baseCommit *apistructs.Commit
+				if tempBranch != "" {
+					if err := s.IdempotentCreateBranch(ctx, repoPath, v.TargetBranch, tempBranch); err != nil {
+						return err
+					}
+					if v.IsJoinTempBranch {
+						baseCommit, err = s.p.bdl.GetMergeBase(apis.GetUserID(ctx), apistructs.GittarMergeBaseRequest{
```

Comment:
do you need batch get merge base? loop may be very big.

Reply:
Generally, there is only one application